qualtran/cirq_interop/_cirq_to_bloq.py

```diff
@@ -300,29 +300,80 @@ def _gather_input_soqs(
     return qvars_in
 
 
-def _extract_bloq_from_op(op: 'cirq.Operation') -> Bloq:
-    """Get a `Bloq` out of a cirq Operation.
-
-    Unwrap BloqAsCirqGate, pass through any GateWithRegisters, and wrap
-    true cirq gates with `CirqGateAsBloq`.
-    """
+def _cirq_gate_to_bloq(gate: cirq.Gate) -> Bloq:
+    from qualtran import Adjoint
+    from qualtran.bloqs.basic_gates import (
+        CNOT,
+        CZPowGate,
+        Hadamard,
+        Rx,
+        Ry,
+        Rz,
+        TGate,
+        Toffoli,
+        TwoBitSwap,
+        XGate,
+        XPowGate,
+        YPowGate,
+        ZGate,
+        ZPowGate,
+    )
+    from qualtran.cirq_interop import CirqGateAsBloq
     from qualtran.cirq_interop._bloq_to_cirq import BloqAsCirqGate
 
-    if op.gate is None:
-        raise ValueError(f"Only gate operations are supported, not {op}.")
-
-    gate = op.gate
     if isinstance(gate, BloqAsCirqGate):
         # Perhaps this operation was constructed from `Bloq.on()`.
         return gate.bloq
     if isinstance(gate, Bloq):
         # I.e., `GateWithRegisters`.
         return gate
 
-    # A base cirq gate.
+    if gate == cirq.T:
```

Comment:
Is it worth using a dictionary for O(1) lookup here?

Reply:
This is pulled out in https://github.com/quantumlib/Qualtran/pull/746. We can continue the discussion there but Tl;Dr is we also need isinstance checks below but we can definitely replace the first few equality checks with a dictionary lookup. Happy to do it as part of the other PR